tests/functional/event_handler/test_openapi_security.py

```diff
@@ -26,31 +30,67 @@ def handler():
 
 
 def test_openapi_top_level_security_missing():
+    # GIVEN an APIGatewayRestResolver instance
     app = APIGatewayRestResolver()
 
     @app.get("/")
     def handler():
         raise NotImplementedError()
 
+    # WHEN the get_openapi_schema method is called with security defined without security schemes
+    # THEN a ValueError should be raised
     with pytest.raises(ValueError):
         app.get_openapi_schema(
             security=[{"apiKey": []}],
         )
 
 
 def test_openapi_operation_security():
+    # GIVEN an APIGatewayRestResolver instance
     app = APIGatewayRestResolver()
 
     @app.get("/", security=[{"apiKey": []}])
     def handler():
         raise NotImplementedError()
 
+    # WHEN the get_openapi_schema method is called with security defined at the operation level
     schema = app.get_openapi_schema(
         security_schemes={
             "apiKey": APIKey(name="X-API-KEY", description="API Key", in_=APIKeyIn.header),
         },
     )
 
+    # THEN the resulting schema should have security defined at the operation level, not the top level
+    security = schema.security
+    assert security is None
+
+    operation = schema.paths["/"].get
+    security = operation.security
+    assert security is not None
+
+    assert len(security) == 1
+    assert security[0] == {"apiKey": []}
```

Comment:
```suggestion
    top_level_security = schema.security
    path_level_security = schema.paths["/"].get.security

    assert top_level_security is None
    assert path_level_security[0] == {"apiKey": []}
```

* tip: careful with shadowing variables with the same name. 
* tip: when you have multiple assertion, often times you might be able to narrow down to one or two, otherwise split tests. For example, if you're asserting operation level security scheme for a given value, you are already implicitly asserting that `is not None`, `it has at least 1 item` - you can therefore keep it down to a single assert.

Reply:
Thanks a lot for this tips, @heitorlessa! I fixed this here and in another test.